### ISSUE TITLE
Issue with Extra Utils 2

### ISSUE THREAD

**Issue by dafman4** [Epoxide-Software/Enchanting-Plus#114]
I dont know if this is an issue with your mod or Extra Utils 2, If you create an item that required Experience levels in Extra Utils 2 (Ex. enchanted wood) player levels update, but the experience "available" to the player (within the enhanced crafting table) doesn't update. I believe that this leads to a false read on what you can enchant which, when attempting to enchant something you don't actually have the experience for it freezes the game and doesn't appear to come back.
 The first picture you can see my levels and the experience
![2018-09-20_13 49 01](https://user-images.githubusercontent.com/4055370/45841012-2ee01880-bcde-11e8-9252-8eb0586c89e8.png)
the second is an example of the crafting recipe using experience levels
![2018-09-20_13 49 11](https://user-images.githubusercontent.com/4055370/45841022-33a4cc80-bcde-11e8-83fd-cd3d0b489933.png)
followed by a change in levels, but not experience available as well as added enchants that I don't actually have the exp for
![2018-09-20_13 50 47](https://user-images.githubusercontent.com/4055370/45841026-369fbd00-bcde-11e8-91ba-4644b5d734e2.png)
finally the freeze after trying to enchant.
![image](https://user-images.githubusercontent.com/4055370/45841074-4f0fd780-bcde-11e8-82bd-53bd30b1f731.png)

I couldn't find any info in the log, but I've attached it anyway.
[latest.log](https://github.com/Epoxide-Software/Enchanting-Plus/files/2402715/latest.log)

I haven't tried it with other mods that have similar crafting recipes/styles
Sorry for the long post, just hoping to help narrow it down if it's on your side

**Comment by sweetsosweet** [comment]
I don't think this is solely due to Extra Utils 2, as I don't have it but I'm getting a freeze. I enchanted an item which took me from level 37 to level 33, which seemed a little low to me, but I figured this mod was just very generous. Trying to enchant a second item, it says my available xp is 5301, and the cost is 2188, but when I clicked enchant, the game froze. I also don't see anything in my log.

**Comment by darkhax** [comment]
Thanks for the report, your issue has helped me identify an endless loop when removing experience. If they player has an unexpected experience value, like less than the cost, it will try to take exp away from you forever. This is why you freeze, and why your EXP will be 0 when you log back in. Fixing this issue is easy enough, and I already have a fix that will stop the freezes. 

Unfortunately, as @sweetsosweet has commented, there does seem to be an issue with calculating the amount of experience the player has. In the game code there is a "experienceTotal" value which represents how much EXP the player has, however many mods, and parts of vanilla completely ignore updating this value, which is why it is sometimes incorrect. There are ways to fix this as well, however it will take a little bit longer to fix.

**Comment by sweetsosweet** [comment]
Yay, thanks for the quick fix! As you've no doubt noticed from the curse comments, everyone's crazy excited to have this mod back, so thanks so much for everything you've done~

**Comment by dafman4** [comment]
Thanks for being so quick, means a lot! Keep up the good work!
